### ISSUE TITLE
SC3037: builtin vs external command echo

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3376]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC3037
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
echo -e "builtin"
/usr/bin/echo -e "not builtin"
env -S echo -e "not builtin"
```

#### Here's what shellcheck currently says:

--shell=sh:

```
-:1:6: warning: In POSIX sh, echo flags are undefined. [SC3037]
-:2:15: warning: In POSIX sh, echo flags are undefined. [SC3037]
```

--shell=bash: nothing

#### Here's what I wanted or expected to see:

- the first line calls a shell builtin, and shellcheck works as expected
- the second line calls an external program, therefore SC3037 should be issued independently of shell
    - GNU coreutils supports `-n`, `-e` and `-E`, while BSD supports only `-n`
- the third line calls an external program via `env`, which is not checked
